### ISSUE TITLE
Support arrays and booleans within JSON?

### ISSUE THREAD

**Issue by mctully** [Graylog2/gelfclient#29]
Is it intentional that arrays and booleans are not supported by this lib? Eg passing additional fields such as:

```
// lBuild is a GelfMessageBuilder
String[] myStringArray = {"a","b","c"};
Boolean myBool = true;
lBuilder.addAdditionalField("extra",myStringArray);
lBuilder.addAdditionalField("extra2",myBool);
```

Results in the array, for example, being casted to a string and thus coming through as "[Ljava.lang.String;@52f9da27'" etc

The encoder appears to handle numbers, null and string - could it perhaps handle array and boolean too?

**Comment by joschi** [comment]
@mctully Arrays/lists are intentionally not supported because they're not part of the [GELF specification](http://docs.graylog.org/en/2.2/pages/gelf.html).

 Booleans should be supported, though. What's the issue you see with booleans?

**Comment by joschi** [comment]
@mctully Thanks for reporting this bug!

I've fixed the case with booleans in #30.

**Comment by mctully** [comment]
Ok, thanks for your reply and the boolean fix. I'll move away from arrays then. Thanks!
